src/containers/CheckoutPage/CheckoutPage.js

```diff
@@ -237,17 +226,31 @@ export class CheckoutPageComponent extends Component {
     this.setState({ submitting: true });
 
     const { history, speculatedTransaction, currentUser, paymentIntent, dispatch } = this.props;
-    const { stripe, card, message } = values;
+    const { stripe, card, message, formValues } = values;
+    const { name, addressLine1, addressLine2, postal, city, state, country } = formValues;
+
+    const billingDetails = {
+      name,
+      address: {
+        city: city,
+        country: country,
+        line1: addressLine1,
+        line2: addressLine2,
+        postal_code: postal,
+        state: state,
+      },
+      email: ensureCurrentUser(currentUser).attributes.email,
+    };
+
+    debugger;
 
     const requestPaymentParams = {
       pageData: this.state.pageData,
       speculatedTransaction,
       stripe,
       card,
-      cardholderName: 'John Doe', // TODO get cardholder's name from form and billing address too
-      email: ensureCurrentUser(currentUser).attributes.email,
+      ...billingDetails,
```

Comment:
Don't spread this. You are using it in `handlePaymentIntent` as  `billingDetails`